### ISSUE TITLE
执行train.py报错

### ISSUE THREAD

**Issue by anmingan** [Aria-Zhangjl/StoryWeaver#5]

**Comment by anmingan** [comment]
您好，执行sample.py时报错

环境;windows:

命令:python sample.py   --pretrained_model_name_or_path=D:\StoryWeaver\models\sd-v1-5    --dataset_dir=D:\StoryWeaver\TBC_Bench\eval-plot\eval_new_story\Frozen\Single\Anna\1.txt   --resolution=512   --output_dir=D:\StoryWeaver\output  --seed=42   --max_char=500  

报错：

![Image](https://github.com/user-attachments/assets/b705beee-9057-4046-9819-552dc0c5acdd)

本地模型：

![Image](https://github.com/user-attachments/assets/99b2fed1-5250-4397-a6a5-dd23ae7ba097)

感谢

**Comment by Aria-Zhangjl** [comment]
Hi, 
Sorry for the late response. 
To generate story scenes, first you have to train your the customized model of each story world using the training scripts. Additionally, in Windows, when specifying file paths, you need to use double backslashes \\ instead of a single backslash \. This is because the backslash character \ is an escape character in many programming languages and in Windows file paths. For example, if you want to specify a file path like D:\StoryWeaver\TBC_Bench\eval-plot\eval_new_story\Frozen\Single\Anna\1.txt, you need to represent it as D:\\\StoryWeaver\\\TBC_Bench\\\eval-plot\\\eval_new_story\\\Frozen\\\Single\\\Anna\\\1.txt in your code.

**Comment by anmingan** [comment]
您好，

当我想要使用demo生成时，第一步需要训练模型。但是执行train.py时遇到一些错误

执行命令：

![Image](https://github.com/user-attachments/assets/5549a2ad-06e0-4a5e-aa89-1e08b78cbfea)

报错：![Image](https://github.com/user-attachments/assets/46edc1eb-4d13-4760-8add-510b7e1189ec)

我用的conda管理虚拟环境。是否和依赖版本有关系呢

感谢！

**Comment by Aria-Zhangjl** [comment]
> 您好，
> 
> 当我想要使用demo生成时，第一步需要训练模型。但是执行train.py时遇到一些错误
> 
> 执行命令：
> 
> ![Image](https://github.com/user-attachments/assets/5549a2ad-06e0-4a5e-aa89-1e08b78cbfea)
> 
> 报错：![Image](https://github.com/user-attachments/assets/46edc1eb-4d13-4760-8add-510b7e1189ec)
> 
> 我用的conda管理虚拟环境。是否和依赖版本有关系呢
> 
> 感谢！

Hi,
I have updated the code for the dataset. You can have a try. :relaxed:

**Comment by anmingan** [comment]
![Image](https://github.com/user-attachments/assets/954cce27-0064-4b24-9303-602c7d8d107c)

还是这个错误  🤣

>python train.py --pretrained_model_name_or_path=D:\\StoryWeaver\\models\\sd-v1-5  --dataset_dir=D:\\StoryWeaver\\TBC_Bench\\train-folder\\Pororo_the_Little_Penguin --test_dataset_dir=D:\\StoryWeaver\\TBC_Bench\\eval-plot\\eval_train\each\\Pororo_the_Little_Penguin --resolution=512 --center_crop --random_flip  --train_batch_size=4 --gradient_accumulation_steps=1 --gradient_checkpointing --checkpointing_steps=200 --max_train_steps=2000 --learning_rate=7e-06  --max_grad_norm=1 --lr_scheduler="constant" --lr_warmup_steps=0 --output_dir=output/Pororo --seed=42 --max_char=2

**Comment by anmingan** [comment]
![Image](https://github.com/user-attachments/assets/f926a6ff-8b66-42b1-ad5c-15ee3219610b)

还是这个错误  🤣

python train.py --pretrained_model_name_or_path=D:\\StoryWeaver\\models\\sd-v1-5  --dataset_dir=D:\\StoryWeaver\\TBC_Bench\\train-folder\\Pororo_the_Little_Penguin --test_dataset_dir=D:\\StoryWeaver\\TBC_Bench\\eval-plot\\eval_train\\each\\Pororo_the_Little_Penguin --resolution=512 --center_crop --random_flip  --train_batch_size=4 --gradient_accumulation_steps=1 --gradient_checkpointing --checkpointing_steps=200 --max_train_steps=2000 --learning_rate=7e-06  --max_grad_norm=1 --lr_scheduler="constant" --lr_warmup_steps=0 --output_dir=output/Pororo --seed=42 --max_char=2

**Comment by Aria-Zhangjl** [comment]
> ![Image](https://github.com/user-attachments/assets/f926a6ff-8b66-42b1-ad5c-15ee3219610b)
> 
> 还是这个错误 🤣
> 
> python train.py --pretrained_model_name_or_path=D:\StoryWeaver\models\sd-v1-5 --dataset_dir=D:\StoryWeaver\TBC_Bench\train-folder\Pororo_the_Little_Penguin --test_dataset_dir=D:\StoryWeaver\TBC_Bench\eval-plot\eval_train\each\Pororo_the_Little_Penguin --resolution=512 --center_crop --random_flip --train_batch_size=4 --gradient_accumulation_steps=1 --gradient_checkpointing --checkpointing_steps=200 --max_train_steps=2000 --learning_rate=7e-06 --max_grad_norm=1 --lr_scheduler="constant" --lr_warmup_steps=0 --output_dir=output/Pororo --seed=42 --max_char=2

我重新update了一下，现在咧？

**Comment by anmingan** [comment]
您好，参数类型type不报错了，现在有个文件错误，

![Image](https://github.com/user-attachments/assets/5a6617bd-d227-421c-aaee-c6259d486e53)

执行参数python train.py  --pretrained_model_name_or_path=D:\\StoryWeaver\\models\\sd-v1-5  --dataset_dir=D:\\StoryWeaver\\TBC_Bench\\train-folder\\Pororo_the_Little_Penguin --test_dataset_dir=D:\\StoryWeaver\\TBC_Bench\\eval-plot\\eval_train\\each\\Pororo_the_Little_Penguin --resolution=512 --center_crop --random_flip  --train_batch_size=4 --gradient_accumulation_steps=1 --gradient_checkpointing --checkpointing_steps=200 --max_train_steps=2000 --learning_rate=7e-06  --max_grad_norm=1 --lr_scheduler="constant" --lr_warmup_steps=0 --output_dir=output/Pororo --seed=42 --max_char=2

是我参数写的不对嘛，感谢感谢

**Comment by Aria-Zhangjl** [comment]
> 您好，参数类型type不报错了，现在有个文件错误，
> 
> ![Image](https://github.com/user-attachments/assets/5a6617bd-d227-421c-aaee-c6259d486e53)
> 
> 执行参数:python train.py --pretrained_model_name_or_path=D:\StoryWeaver\models\sd-v1-5 --dataset_dir=D:\StoryWeaver\TBC_Bench\train-folder\Pororo_the_Little_Penguin --test_dataset_dir=D:\StoryWeaver\TBC_Bench\eval-plot\eval_train\each\Pororo_the_Little_Penguin --resolution=512 --center_crop --random_flip --train_batch_size=4 --gradient_accumulation_steps=1 --gradient_checkpointing --checkpointing_steps=200 --max_train_steps=2000 --learning_rate=7e-06 --max_grad_norm=1 --lr_scheduler="constant" --lr_warmup_steps=0 --output_dir=output/Pororo --seed=42 --max_char=2
> 
> 是我参数写的不对嘛，感谢感谢

路径的问题，windows系统下文件路径中的单斜杠\要写成双斜杠\\

**Comment by anmingan** [comment]
![Image](https://github.com/user-attachments/assets/1bcd5c68-4443-4cd3-91f9-3d1c1a975ad2)

参数命令中路径是双斜杠

另外在我本地中 ，D:\desktop\StoryWeaver\TBC_Bench\train-folder\Pororo_the_Little_Penguin\image\Poby   这里面只有一些图片，没有5.txt ，请问这个5.txt是执行脚本的时候生成的嘛？

![Image](https://github.com/user-attachments/assets/2ed522bb-1c59-4d86-b768-c66b27be32e2)

**Comment by Aria-Zhangjl** [comment]
> ![Image](https://github.com/user-attachments/assets/1bcd5c68-4443-4cd3-91f9-3d1c1a975ad2)
> 
> 参数命令中路径是双斜杠
> 
> 另外在我本地中 ，D:\desktop\StoryWeaver\TBC_Bench\train-folder\Pororo_the_Little_Penguin\image\Poby 这里面只有一些图片，没有5.txt ，请问这个5.txt是执行脚本的时候生成的嘛？
> 
> ![Image](https://github.com/user-attachments/assets/2ed522bb-1c59-4d86-b768-c66b27be32e2)

原本TBC_dataset.py代码316行有关于文本路径的设置：

<img width="1060" alt="Image" src="https://github.com/user-attachments/assets/3841c940-442a-46d0-ba89-6d6d7b1f1574" />

文本均放置于相同角色文件夹的caption文件夹中

**Comment by anmingan** [comment]
很奇怪，

caption中有对应的文本，但是读取不到路径 

![Image](https://github.com/user-attachments/assets/e704b422-e170-4285-a478-372e5d5898d3)

这个路径中有四个反斜杠 请问这是正确的嘛

**Comment by Aria-Zhangjl** [comment]
> 很奇怪，
> 
> caption中有对应的文本，但是读取不到路径
> 
> ![Image](https://github.com/user-attachments/assets/e704b422-e170-4285-a478-372e5d5898d3)
> 
> 这个路径中有四个反斜杠 请问这是正确的嘛

不对的，还是路径的问题

**Comment by anmingan** [comment]
> > 很奇怪，
> > caption中有对应的文本，但是读取不到路径
> > ![Image](https://github.com/user-attachments/assets/e704b422-e170-4285-a478-372e5d5898d3)
> > 这个路径中有四个反斜杠 请问这是正确的嘛
> 
> 不对的，还是路径的问题

您好，

看代码报错好像是，/image 没有替换成/caption

我更改了下代码 

![Image](https://github.com/user-attachments/assets/6fe4bee6-e3ed-4963-9a79-9effda6adee6)

没有报路径的错误了 ，但是有个新问题 ，不知道是否和我更改的代码有关系

![Image](https://github.com/user-attachments/assets/52bcea06-0e18-459c-95d0-23f3fe124d6e)

感谢
